chia/full_node/full_node.py

```diff
@@ -595,7 +595,7 @@ async def short_sync_batch(self, peer: WSChiaConnection, start_height: uint32, t
                 self.sync_store.batch_syncing.remove(peer.peer_node_id)
                 self.log.error(f"Error short batch syncing, could not fetch block at height {start_height}")
                 return False
-            if not self.blockchain.contains_block(first.block.prev_header_hash):
+            if not self.blockchain.contains_block(first.block.prev_header_hash, first.block.height - 1):
```

Comment:
here, we're not really asking whether the blocks i part of the main chain, we're asking if its in the block cache. if it isn't, we declare this a deep reorg or catch-up (and presumably transition to long-sync).

Reply:
your right

Reply:
i disagree, for short_sync_batch we assume we are syncing from start_height where start_height - 1 is in our chain so checking if first -1 is in the main chain is correct 

this is only done for the first_block.prev_header_hash which is expected to be the fork point i.e in out chain

Reply:
right, I guess we expect both in this case.